docs/administration/Parametres de configuration.adoc
= SDIS Remocra - Paramètres de configuration

ifdef::env-github,env-browser[:outfilesuffix: .adoc]

:experimental:
:icons: font

:toc:

:numbered:

link:../index{outfilesuffix}[Retour à l'accueil]

link:../Manuel%20administration{outfilesuffix}[Retour au manuel d'administration]

'''

Les paramètres dont la clé commence par ```PDI_``` concernent a priori Pentaho Data Integration.


== Authentification depuis un autre service ==

L'authentification LDAP est prioritaire sur l'authentification à partir de la base de données. Le couple identiant / mot de passe saisi par un utilisateur peut être vérifié selon deux méthodes :

* *Simple*
** L'authentification est réalisée directement avec le couple identifiant / mot de passe fourni. C'est possible lorsque le ```CN``` de l'utilisateur LDAP est le même que le ```remocra.utilisateur.identifiant``` de la base de données.
** Exemple de configuration (les autres paramètres ```\*\_LDAP_*``` sont laissés vides) :
*** Accès au serveur :
**** ```PDI_LDAP_URL_HOST``` : ```atolcd-sdis77-ad.cloud.priv.atolcd.com```
**** ```PDI_LDAP_URL_PORT``` : ```389```
**** ```PDI_LDAP_URL_BASE_DN``` : ``````


* *Recherche de l'utilisateur*
** Une première requête avec un compte administrateur permet de retrouver le ```CN``` de l'utilisateur. C'est le couple ```CN``` / mot de passe saisi qui est utilisé pour valider l'authentification.
** Exemple de configuration:
*** Accès au serveur :
**** ```PDI_LDAP_URL_HOST``` : ```atolcd-sdis77-ad.cloud.priv.atolcd.com```
**** ```PDI_LDAP_URL_PORT``` : ```389```
**** ```PDI_LDAP_URL_BASE_DN``` : ``````
*** Compte administrateur :
**** ```PDI_LDAP_ADMIN_DN``` : ```cn=Administrator,cn=Users,dc=sdis77-ad,dc=local```
**** ```PDI_LDAP_ADMIN_PASSWORD``` : ```lemotdepasse```
*** Recherche de l'utilisateur :
**** ```PDI_LDAP_USER_BASE_NAME``` : ```ou=Sites SDIS 77,dc=sdis77-ad,dc=local```
**** ```PDI_LDAP_USER_FILTER``` : ```(&(objectclass=user)(sAMAccountName=[USERNAME]))```

=== PDI_LDAP_ADMIN_DN ===
* Distinguished Name du compte admin
* Obligatoire pour activer la méthode "recherche d''utilisateur"
* Exemple : ```cn=Administrator,cn=Users,dc=sdis77-ad,dc=local```

=== PDI_LDAP_ADMIN_PASSWORD ===
* Mot de passe du compte admin
* Obligatoire pour activer la méthode "recherche d'utilisateur"

=== PDI_LDAP_URL_BASE_DN ===
* LDAP : Distinguished Name de base : ldap://hôte:port/<b>dn</b>
* Facultatif
* * Exemple : ```cn=Users,dc=sdis,dc=fr```

=== PDI_LDAP_URL_HOST ===
* LDAP : Hôte du serveur : ldap://<b>hôte</b>:port/dn
* Obligatoire pour activer l'authentification LDAP
* Exemple : ```ldap.sdisXX.fr```

=== PDI_LDAP_URL_PORT ===
* LDAP : Port du serveur : ldap://hôte:<b>port</b>/dn
* Par défaut : ```389```

=== PDI_LDAP_USER_BASE_NAME ===
* Base name des comptes utilisateurs
* Obligatoire pour activer la méthode "recherche d'utilisateur"
* Exemple : ```ou=Sites SDIS 77,dc=sdis77-ad,dc=local```

=== PDI_LDAP_USER_FILTER ===
* Filtre des comptes utilisateurs
* ```[USERNAME]``` est automatiquement remplacé par lidentifiant de l'utilisateur.
* Par défaut : ```(&(objectclass=user)(sAMAccountName=[USERNAME]))```


== Authentification vers un autre service ==

=== SORTIE_JWT_CLEPRIVEE ===
* Clé privée (.der) pour les jetons jwt

=== SORTIE_JWT_CLEPUBLIQUE ===
* Clé publique (.der) pour les jetons jwt

=== SORTIE_JWT_ISSUER ===
* Service générateur des jetons jwt (remocra par défaut)
* Par défaut : ```remocra```

=== SORTIE_JWT_VALIDITE_SEC ===
* Durée de validité des jetons jwt en secondes (30 par défaut)
* Par défaut : ```30```


== Cartographie ==

=== CLES_IGN
* Clé(s) IGN. Exemples (mono ou multi)
* Exemple : ```rm5nr4ejwapq6d58fm0869bd```
* Exemple :
[source,js]
----
{
  'remocra.sdisxx.fr': '9s507j21zeha5rp5pkll48od',
  'remocra-preprod.sdisxx.fr': 'dds5nr4ejwapq6d58fm086c7d'
}
----


=== COORDONNEES_FORMAT_AFFICHAGE
* Expression des coordonnées (valeurs admises : DD_DDDD, DD_MM_MM, DD_MM_SSSS)
* Par défaut : ```DD_DDDD```

=== DEFAULT_ORIENTATION_X
* Orientation par défaut en longitude (saisie des coordonnées) : E ou O
* Par défaut : ```E```

=== DEFAULT_ORIENTATION_Y
* Orientation par défaut en latitude (saisie des coordonnées) : N ou S
* Par défaut : ```N```

=== WMS_BASE_URL
* URL de base complète du serveur de cartes
* Par défaut : ```http://localhost:8090/geoserver```

=== WMS_PUBLIC_LAYERS
* Couches publiques du serveur WMS avec séparateur %. Exemple : remocra:ADMINISTRATIF%remocra:RISQUE



== Chemins sur disque ==

=== DOSSIER_COURRIER ===
* Emplacement du dossier de stockage des courriers
* Par défaut : ```/var/remocra/courriers```

=== DOSSIER_DEPOT_ALERTE ===
* Emplacement du dossier de stockage des alertes
* Par défaut : ```/var/remocra/alertes```

=== DOSSIER_DEPOT_ATTESTATION ===
* Emplacement du dossier de stockage des attestations de débit simultané
* Par défaut : ```/var/remocra/attestations```

=== DOSSIER_DEPOT_BLOC ===
* Emplacement du dossier de stockage des documents des blocs
* Par défaut : ```/var/remocra/blocs```

=== DOSSIER_DEPOT_DECLA_HYDRANT ===
* Emplacement du dossier de stockage des déclarations d'hydrants
* Par défaut : ```/var/remocra/declahydrant```

=== DOSSIER_DEPOT_DELIB ===
* Emplacement du dossier de stockage des délibérations
* Par défaut : ```/var/remocra/deliberations```

=== DOSSIER_DEPOT_PERMIS ===
* Emplacement du dossier de stockage des permis
* Par défaut : ```/var/remocra/permis```

=== DOSSIER_DEPOT_RCI ===
* Emplacement du dossier de stockage des documents RCI
* Par défaut : ```/var/remocra/rci```

=== DOSSIER_DEPOT_RECEP_TRAVAUX ===
* Emplacement du dossier de stockage des dossiers de réception de travaux
* Par défaut : ```/var/remocra/receptravaux```

=== DOSSIER_DOC_HYDRANT ===
* Emplacement du dossier de stockage des photos des hydrants
* Par défaut : ```/var/remocra/hydrants```

=== DOSSIER_DOC_OLDEBVISITE ===
* Emplacement du dossier de stockage des documents de la visite
* Par défaut : ```/var/remocra/oldebvisite```

=== DOSSIER_GETFEATUREINFO ===
* Emplacement du dossier des transformations GetFeatureInfo
* Par défaut : ```/var/remocra/getfeatureinfo```

=== DOSSIER_LAYERS ===
* Emplacement du dossier de configuration des couches (layers.json)
* Par défaut : ```/var/remocra/layers```

=== DOSSIER_RESSOURCES_EXTERNES ===
* Emplacement du dossier de stockage des ressources externes (pages accueil, cartes, images des légendes, etc.)
* Par défaut : ```/var/remocra/html```

=== PDI_CHEMIN_KML ===
* Dossier de stockage du fichier KML des risques technologiques
* Par défaut : ```/var/remocra/pdi/kml```

=== PDI_CHEMIN_LOG ===
* Dossier de stockage des fichiers de trace de l'ETL Pentaho Data Integration
* Par défaut : ```/var/remocra/pdi/log```

=== PDI_CHEMIN_MODELES ===
* Dossier de stockage des modèles de documents utilisés par l'ETL Pentaho Data Integration
* Par défaut : ```/var/remocra/modeles```

=== PDI_CHEMIN_SYNCHRO ===
* Dossier de stockage temporaire des fichiers d'échanges APIS – REMOCRA
* Par défaut : ```/var/remocra/pdi/synchro```

=== PDI_CHEMIN_TMP ===
* Dossier de travail temporaire de l'ETL Pentaho Data Integration
* Par défaut : ```/var/remocra/pdi/tmp```

=== PDI_CHEMIN_TRAITEMENT ===
* Dossier de stockage des documents produits par l'ETL Pentaho Data Integration à proposer en téléchargement
* Par défaut : ```/var/remocra/pdi/export```

=== PDI_DOSSIER_DEPOT ===
* Emplacement du dossier de stockage des fichiers de PDI
* Par défaut : ```/var/remocra/pdi/depot```

=== PDI_FICHIER_PARAMETRAGE ===
* Chemin et nom complet du fichier de configuration de l'ETL Pentaho Data Integration
* Par défaut : ```/home/postgres/remocra_pdi/remocra.properties```


== Courriels et courriers ==

=== EMAIL_DEST_CREATION_RCI ===
* Adresse mél utilisée pour la diffusion lorsqu'un départ de feux est créé
* Exemple : ```rcci@sdisxx.fr```

=== EMAIL_DEST_DEPOT_DECLAHYDRANT ===
* Adresse du destinataire des emails de notification de dépôts de dossiers de déclaration d'hydrant(s)
* Exemple : ```pei-decla@sdisxx.fr```

=== EMAIL_DEST_DEPOT_DELIB ===
* Adresse du destinataire des emails de notification de dépôts de délibérations
* Exemple : ```delib@sdisxx.fr```

=== EMAIL_DEST_DEPOT_RECEPTRAVAUX ===
* Adresse du destinataire des emails de notification de dépôts de dossiers de réception de travaux
* Exemple : ```reception-travaux@sdisxx.fr```

=== PDI_NOTIFICATION_GENERAL_UTILISATEUR_ID ===
* Identifiant de l'utilisateur à notifier dans le cadre de la notification générale (erreurs générales, autres, etc.)
* Exemple : ```5```

=== PDI_NOTIFICATION_KML_UTILISATEUR_ID ===
* Identifiant de l'utilisateur à notifier dans le cas de la publication d'un nouveau fichier de risques KML
* Exemple : ```6```

=== PDI_PDF_SIGN_KEY_PASSWORD ===
* Signature des courriers PDF : Mot de passe de la clé (pas de signature si vide)

=== PDI_PDF_SIGN_PFX_FILE ===
* Signature des courriers PDF : Emplacement du fichier PFX de signature des courriers PDF (pas de signature si vide)

=== PDI_PDF_SIGN_PFX_PASSWORD ===
* Signature des courriers PDF : Mot de passe d'accès au contenu du fichier PFX (pas de signature si vide)

=== PDI_SMTP_AR_MAIL ===
* Adresse mél utilisée pour recueillir les accusés de réception des messages envoyés par Remocra
* Exemple : ```remocra-ar@sdisxx.fr```

=== PDI_SMTP_EME_MAIL ===
* Adresse mél d'expédition utilisée pour l'envoi de messages par Remocra
* Exemple : ```remocra@sdisxx.fr```

=== PDI_SMTP_EME_NAME ===
* Nom de l'expediteur utilisé pour l'envoi de messages par Remocra
* Par défaut : ```Application Remocra```

=== PDI_SMTP_ERR_MAIL ===
* Adresse mél utilisée pour recueillir les erreurs de non délivrance des messages envoyés par Remocra
* Exemple : ```remocra@sdisxx.fr```

=== PDI_SMTP_PASSWORD ===
* Mot de passe du serveur SMTP utilisé pour l'envoi de messages par Remocra

=== PDI_SMTP_PORT ===
* Numéro du port du serveur SMTP utilisé pour l'envoi de messages par Remocra
* Par défaut : ```25```

=== PDI_SMTP_REP_MAIL ===
* Adresse mél de réponse utilisée pour l'envoi de messages par Remocra
* Exemple : ```no-reply@sdisxx.fr```

=== PDI_SMTP_URL ===
* URL du serveur SMTP utilisé pour l'envoi de messages par Remocra
* Exemple : ```localhost```

=== PDI_SMTP_USER ===
* Nom d'utilisateur du serveur SMTP utilisé pour l'envoi de messages par Remocra

=== PDI_UTILISATEUR_NOTIF_INDISPO ===
* Code du profil ou adresse du destinataire pour la notification de début ou de fin prévisionnelles des indisponibilités temporaires
* Exemple : ```SDIS-REF-DECI```
* Exemple : ```deci@sdisxx.fr```

== Général ==

=== COMMUNES_INSEE_LIKE_FILTRE_SQL ===
* Filtre SQL pour la récupération des communes. Exemple : "83%"
* Il s'applique pour le choix des communes et la recherche suite à un clic (permis, etc.).
* Par défaut : ```83%```

=== COMPLEXITE_PASSWORD ===
* Configuration de la complexité du mot de passe des utilisateurs
* Par défaut : ```libre```
* Valeurs possibles:
** ```libre``` Mot de passe libre
** ```complexe``` 9 caractères minimum, doit contenir au moins 1 lettre et 1 chiffre

=== MENTION_CNIL ===
* Mention CNIL affichée dans le pied de page
* Par défaut : ``````

=== MESSAGE_ENTETE ===
* Texte affiché dans l'entête du site
* Par défaut : ``````

=== TITRE_PAGE ===
* Titre de la page
* Par défaut : ```SDIS - REMOcRA```

=== PDI_CODE_SDIS ===
* Code du département
* Exemple : `83`

=== PDI_URL_SITE ===
* URL de base complète du site Remocra (Commence par "http" et finit juste avant le #)
* Exemple : ```https://remocra.sdisxx.fr/remocra/```

=== TOLERANCE_VOIES_METRES ===
* Tolérance de chargement des voies, exprimée en mètres
* Par défaut : ```150```

== Métadonnées ==

=== PDI_METADATA_FILTRE_CQL ===
* Filtre à utiliser pour la récupération des métadonnées
** Exemple : ```OrganisationName like '%SDIS%'```

=== PDI_METADATA_FILTRE_MAX ===
* Nombre d'enregistrements maximum à retourner
** Exemple : ```20```

=== PDI_METADATA_URL_FICHE_COMPLETE ===
* URL d'accès aux fiches de métadonnées HTML du CRIGE PACA
** Exemple : ```http://www.crige-paca.org/carto/fonctionnalites/geocatalogue/geosource.php```

=== PDI_METADATA_URL_GEOCATALOGUE ===
* URL d'accès au géocatalogue du CRIGE PACA
** Exemple : ```http://geocatalogue.crige-paca.org/geonetwork```


== Permis ==

=== PERMIS_TOLERANCE_CHARGEMENT_METRES ===
* Tolérance de chargement des permis, exprimée en mètres
* Par défaut : ```1000```



== Points d'eau ==

=== CITERNE_TOLERANCE_ASSOCIATION_PI_METRES ===
* Tolérance de chargement des PI associables aux Citernes, exprimée en mètres
* Par défaut : ```500```

=== HYDRANT_DELAI_CTRL_URGENT ===
* Nombre de jours avant échéance où un contrôle est considéré comme "urgent"
* Par défaut : ```15```

=== HYDRANT_DELAI_CTRL_WARN ===
* Nombre de jours avant échéance où un contrôle est considéré comme "à faire bientôt"
* Par défaut : ```45```

=== HYDRANT_DELAI_RECO_URGENT ===
* Nombre de jours avant échéance où une reconnaisance est considérée comme "urgente"
* Par défaut : ```15```

=== HYDRANT_DELAI_RECO_WARN ===
* Nombre de jours avant échéance où une reconnaisance est considérée comme "à faire bientôt"
* Par défaut : ```45```

=== HYDRANT_NUMEROTATION_INTERNE_METHODE ===
* Règle de calcul des <i>numéros internes</i> des PEI (ex : 77, 83)
* Par défaut : ```83```

Méthodes disponibles pour les règles de calcul des numéro internes et globaux : link:../pei/Numérotation_PEI{outfilesuffix}[Numérotation des PEI]

=== HYDRANT_NUMEROTATION_METHODE ===
* Règle de calcul des <i>numéros globaux</i> des PEI (ex : 77, 83)
* Par défaut : ```83```

Méthodes disponibles pour les règles de calcul des numéro internes et globaux : link:../pei/Numérotation_PEI{outfilesuffix}[Numérotation des PEI]

=== HYDRANT_RENUMEROTATION_ACTIVATION ===
* Activer la renumérotation automatique lors de la mise à jour
* Par défaut : ```true```

=== HYDRANT_RENOUVELLEMENT_CTRL_PUBLIC ===
* Délai légal entre 2 contrôles techniques périodiques sur PEI de DECI publique ou privée sous convention (en jours)
* Par défaut : ```365```

=== HYDRANT_RENOUVELLEMENT_CTRL_PRIVE ===
* Délai légal entre 2 contrôles techniques périodiques sur PEI de DECI privée (en jours)
* Par défaut : ```365```

=== HYDRANT_RENOUVELLEMENT_RECO_PUBLIC ===
* Délai légal entre 2 reconnaissances opérationnelles périodiques sur PEI de DECI publique ou privée sous convention (en jours)
* Par défaut : ```180```

=== HYDRANT_RENOUVELLEMENT_RECO_PRIVE ===
* Délai légal entre 2 reconnaissances opérationnelles périodiques sur PEI de DECI privée (en jours)
* Par défaut : ```365```

=== HYDRANT_SYMBOLOGIE_METHODE ===
* Symbologie des PEI (ex : WMS, GEN, 42, 77, 83, 89)
* Par défaut : ```GEN```

Méthodes disponibles pour la symbologie des PEI : link:../pei/Symbologie_PEI{outfilesuffix}[Symbologies disponibles]

=== HYDRANT_NOMBRE_HISTORIQUE ===
* Nombre de mesures à prendre en compte dans l'historique des mesures débits/pressions
* Par défaut : ```3```

=== HYDRANT_VISITE_RAPIDE ===
* Activer la saisie rapide de visite (ouverture sur clic)
* Par défaut : ```false```

=== HYDRANT_HIGHLIGHT_DUREE ===
* Durée de mise en évidence des PEI sur la carte en millisecondes. Exemples : 0 pour désactiver la mise en évidence, 1000 pour une seconde
* Par défaut : ```1000```

=== HYDRANT_COLONNES ===
* Colonnes et ordre des colonnes à afficher dans le tableau de suivi des PEI, séparateur %
* Par défaut: ```numero%nomTournee%natureNom%dateReco%dateContr%dispoTerrestre%dispoHbe```
* Valeurs possibles:
** ```numero``` Numéro du PEI
** ```nomTournee``` Tournée à laquelle appartient le PEI
** ```natureNom``` Type du PEI
** ```dateReco``` Date de la prochaine reconnaissance
** ```dateContr``` Date du prochain contrôle
** ```dispoTerrestre``` Disponibilité terrestre
** ```dispoHbe``` Disponibilité des Hélicoptères Bombardiers d'Eau
** ```nomCommune``` Nom de la commune à laquelle appartient le PEI
** ```nomNatureDeci``` Statut de la nature DECI (privé, publique, privé sous convention)
** ```adresse``` Adresse postale
** ```numeroInterne``` Numéro interne du PEI
** ```autoriteDeci``` Autorité de police DECI

=== HYDRANT_LONGUE_INDISPONIBILITE_JOURS ===
* Nombre de jours avant de considérer un PEI comme indisponible depuis trop longtemps
* Le nombre de jours doit être strictement positif. Dans le cas contraire, l'affichage du message est désactivé
* Par défaut : ```180```

=== HYDRANT_LONGUE_INDISPONIBILITE_MESSAGE ===
* Message à afficher en cas de PEI indisponible depuis trop longtemps
* Si laissé vide, l'affichage du message est désactivé pour les utilisateurs
* Le message ne s'affiche que pour les utilisateurs faisant partie d'un type d'organisme ayant pour code ```COMMUNE``` ou ```EPCI```
* Par défaut : ```Un ou plusieurs PEI situé(s) sur votre territoire est/sont indisponible(s) depuis plus de %MOIS% mois et %JOURS% jours :```
** ``%MOIS%`` sera remplacé par le nombre de mois calculé depuis HYDRANT_LONGUE_INDISPONIBILITE_JOURS
** ``%JOURS%`` sera remplacé par le nombre de jours calculé depuis HYDRANT_LONGUE_INDISPONIBILITE_JOURS (modulo le nombre de mois)

=== HYDRANT_LONGUE_INDISPONIBILITE_TYPEORGANISME ===
* Organismes concernés par le message à afficher en cas de PEI indisponible depuis trop longtemps
* Correspond à une expression régulière java
* Par défaut : ```^(COMMUNE|EPCI)$```

=== HYDRANT_ZOOM_NUMERO ===
* Autorise l'utilisateur à effectuer des zoom sur des adresses via l'API adresse.data.gouv.fr
* Par défaut : ```false```

=== HYDRANT_DEPLACEMENT_DIST_WARN ===
* Distance de déplacement minimale pour laquelle afficher un avertissement lors de l''import de Contrôles Techniques Périodiques
* Par défaut : ```10```

=== HYDRANT_GENERATION_CARTE_TOURNEE ===
* Active le module de génération de carte des tournées dans le module Point d'eau
* Le module de génération nécessite qu'une https://geoserver-pdf.readthedocs.io/en/stable/webadmin/data/layergroups.html[aggrégation de couche] nommée ```TOURNEE``` soit configurée sur le serveur Geoserver. Il est recommandé que l'aggrégation de couche contienne a minima :
** Un fond de plan
** La couche des hydrants. Le module enverra l'identifiant de la tournée via le https://docs.geoserver.org/stable/en/user/data/database/sqlview.html#parameterizing-sql-views[VIEWPARAM] ```tournee_id``` permettant de discriminer les hydrants à retourner
* Par défaut : ```false```

=== HYDRANT_METHODE_TRI_ALPHANUMERIQUE ===
* Détermine le tri effectué sur le numéro des PEI au sein de l'application
** Si coché, effectue un tri alphanumérique plus adapté aux SDIS ayant une numérotation basée sur des caractère alphabétiques
** Le tri naturel est plus adapté aux SDIS ayant une numérotation basée sur des caractères numériques
* Par défaut : ```false```

=== NIVEAU_TRACABILITE ===
* Niveau de traçabilité des points d'eau (valeurs admises : organisme , utilisateur)
* Par défaut : ```organisme```

=== PDI_DELTA_NOTIF_INDISPO_DEBUT ===
* Notifier par mél N heures avant le début d'une indisponibilité temporaire
* Par défaut : ```1```

=== PDI_DELTA_NOTIF_INDISPO_FIN ===
* Notifier par mél N heures avant la fin d'une indisponibilité temporaire
* Par défaut : ```1```

=== VITESSE_EAU ===
* Vitesse nominale retenue d'un réseau d'eau
* Par défaut : ```2```


== Risques ==

=== PDI_IMAP_PASSWORD ===
* Mot de passe du serveur IMPAP utilisé pour la récupération du fichier KML des risques technologiques

=== PDI_IMAP_PORT ===
* Numéro du port du serveur IMAP utilisé pour la récupération du fichier KML des risques technologiques
* Exemple : ```993```

=== PDI_IMAP_URL ===
* URL du serveur IMAP utilisé pour la récupération du fichier KML des risques technologiques
* Exemple : ```imaps.sdisxx.fr```

=== PDI_IMAP_USER ===
* Nom d'utilisateur du serveur IMAP utilisé pour la récupération du fichier KML des risques technologiques
* Exemple : ```imapuser@sdisxx.fr```


== Synchro SIG ==

=== PDI_DOSSIER_EXPORT_SDIS ===
* Nom du sous-dossier temporaire dans lequel sont déposés les fichiers d'échanges produits par REMOCRA
* Par défaut : ```EXPORT_SDIS```

=== PDI_DOSSIER_IMPORT_EXTRANET ===
* Nom du sous-dossier temporaire dans lequel sont déposés les fichiers d'échanges récupérés d'APIS
* Par défaut : ```IMPORT_EXTRANET```

=== PDI_FTP_DOSSIER_EXTRANET ===
* Nom du sous-dossier du serveur FTP dans lequel récupérer les fichiers produits par APIS
* Par défaut : ```EXPORT_EXTRANET```

=== PDI_FTP_DOSSIER_SDIS ===
* Nom du sous-dossier du serveur FTP dans lequel déposer les fichiers produits par REMOCRA
* Par défaut : ```IMPORT_SDIS```

=== PDI_FTP_DOSSIER_SYNCHRO ===
* Dossier du serveur FTP dédié à la synchronisation
* Par défaut : ```SYNCHRO```

=== PDI_FTP_PORT ===
* Port du site FTP
* Exemple : ```21```

=== PDI_FTP_URL ===
* URL d'accès au site FTP
* Exemple : ```ftp.sdisxx.fr```

=== PDI_FTP_USER_NAME ===
* Nom d'utilisateur du compte FTP
* Exemple : ```remocra```

=== PDI_FTP_USER_PASSWORD ===
* Mot de passe du compte FTP

=== PDI_NOM_SCHEMA_ORACLE ===
* Nom du schéma Oracle de la base APIS pour génération des instructions SQL
* Exemple : ```APIS_GEO```

=== PDI_POSTGRESQL_NOM_SCHEMA_REFERENTIEL ===
* Nom du schéma Postgresql dans lequel créer et synchroniser les tables de référentiels géographiques
* Exemple : ```sdis_referentiel```

=== PDI_POSTGRESQL_NOM_SCHEMA_REMOCRA ===
* Nom du schéma Postgresql dans lequel créer et synchroniser les tables métier
* Exemple : ```remocra```

=== PDI_POSTGRESQL_NOM_SCHEMA_SYNCHRO ===
* Nom du schéma PostgreSQL temporaire pour import des données APIS
* Exemple : ```sdis_referentiel```

=== PDI_PREVARISC_UTILISATEUR_ID ===
* Identifiant de l'utilisateur utilisé comme créateur des alertes venant de Prevarisc
* Exemple : ```8```

== Traitements et purge ==

=== ID_TRAITEMENT_ATLAS ===
* Traitement de téléchargement de l'Atlas
* Exemple : ```7```

=== ID_TRAITEMENT_HYDRANTS_NON_NUM ===
* Traitement des points d'eau non numérotés
* Exemple : ```15```

=== ID_TRAITEMENT_NB_ALERTES_PAR_UTILISATEUR ===
* Traitement du nombre d'alertes par utilisateur
* Exemple : ```7```

=== ID_TRAITEMENT_OLDEB ===
* Traitement pour l'impression d'une fiche "Obligation de débroussaillement
* Exemple : ```25```

=== ID_TRAITEMENT_PURGE_KML ===
* Traitement de purge de la couche des risques express
* Exemple : ```8```

=== PDI_PURGE_ALERTE_JOURS ===
* Nombre de jours avant suppression des alertes prises en compte
* Par défaut : ```31```

=== PDI_PURGE_MAIL_JOURS ===
* Nombre de jours avant suppression des messages envoyés par Remocra
* Par défaut : ```1```

=== PDI_PURGE_TRAITEMENT_JOURS ===
* Nombre de jours avant suppression des traitements réalisés par Remocra
* Par défaut : ```1```

=== PROCESS_OFFLINE_USER ===

* Identifiant de l'utilisateur utilisé comme créeateur des processus ETl
* Par défaut : ```remocra-adm-app```

== Planification DECI ==

=== DECI_ISODISTANCES ===
* Isodistances à parcourir lors de la création de la couverture hydraulique
* Nombres entiers par ordre croissant, séparés par des virgules
* Par défaut : ```100, 150, 300, 350```

=== DECI_DISTANCE_MAX_PARCOURS ===
* Distance maximum de parcours sur le réseau routier lors du calcul de la couverture hydraulique
* Par défaut : ```300```


remocra/src/main/webapp/js/app/remocra/features/cartographie/Config.js
Ext.require('Ext.form.FieldSet');
Ext.require('Sdis.Remocra.widget.map.LegendTemplate');

Ext.define('Sdis.Remocra.features.cartographie.Config', {
    extend : 'Ext.Panel',
    alias : 'widget.crCartographieConfig',

    title : 'Configuration',

    defaults : {
        style : 'margin-left:20px',
        labelAlign : 'left',
        labelWidth : 180,
        labelSeparator : '',
        width : 400
    },

    legendUrl : BASE_URL+'/../ext-res/js/app/remocra/features/cartographie/data/carte.json',
    //legendHideSliders : true,
    
    items : [ {
        xtype : 'textfield',
        fieldLabel : 'Titre',
        itemId : 'titre',
        value : 'Nouvelle carte'
    }, {
        xtype : 'combo',
        fieldLabel : 'Format',
        itemId : 'format',
        store : new Ext.data.SimpleStore({
            fields : [ 'value', 'display' ],
            data : [ [ 'a4', 'A4' ], [ 'a3', 'A3' ] ]
        }),
        displayField : 'display',
        valueField : 'value',
        forceSelection : true,
        autoSelect : true,
        editable : false,
        value : 'a4'
    }, {
        xtype : 'combo',
        fieldLabel : 'Orientation',
        itemId : 'orientation',
        store : new Ext.data.SimpleStore({
            fields : [ 'value', 'display' ],
            data : [ [ 'landscape', 'Paysage' ], [ 'portrait', 'Portrait' ] ]
        }),
        displayField : 'display',
        valueField : 'value',
        forceSelection : true,
        autoSelect : true,
        editable : false,
        value : 'landscape'
    }, {
        xtype : 'displayfield',
        fieldLabel : 'Couches'
    }, {
        xtype : 'button',
        itemId : 'open',
        text : 'Ouvrir la carte',
        width : 180,
        style : 'margin-top:20px;'
    }, {
        xtype : 'displayfield',
        itemId : 'couches',
        cls : 'simplemaplegend maplegend forceover',
        style : 'margin-left:190px;float:none'
    } ],

    initComponent : function() {
        this.callParent(arguments);
        this.on('afterrender', this.loadMapConfig, this);
    },
    
    loadMapConfig : function() {
        if (this.legendUrl == null) {
            return;
        }
        Ext.Ajax.request({
            url: this.legendUrl,
            method: 'GET',
            scope: this,
            callback: function(options, success, response) {
                if (success !== true) {
                    Ext.Msg.alert('Carte', 'Un problème est survenu lors du chargement de la configuration.');
                    return;
                }
                // On renseigne les données des couches
                this.legendData = Ext.decode(response.responseText);
                this.prepareLegendData();
                this.renderLegend();
            }
        });
    },
    
    /**
     * Préparation du choix des couches : droits sur les couches, etc.
     * Couche permis : nécessite le droit PERMIS.Read 
     * @returns
     */
    prepareLegendData : function() {
        var legendDataPrepared = Ext.clone(this.legendData);
        var legendData = this.legendData;
        // Chaque groupe (à l'envers)
        for (iGrp = legendData.items.length; iGrp > 0; iGrp--) {
            grp = legendData.items[iGrp - 1];
            // Chaque couche (à l'envers)
            for(iLay=grp.items.length ; iLay>0 ; iLay--) {
                var layerDef = grp.items[iLay-1];
                if (layerDef.id=='permisLayer' && !Sdis.Remocra.Rights.hasRight('PERMIS_R')) {
                    // Pas de droit sur cette couche
                    Ext.Array.erase(legendDataPrepared.items[iGrp - 1].items, iLay - 1, 1);
                }
            }
        }
        // On retire les groupes vides
        for (iGrp = legendDataPrepared.items.length; iGrp > 0; iGrp--) {
            if (Ext.isEmpty(legendDataPrepared.items[iGrp - 1].items)) {
                Ext.Array.erase(legendDataPrepared.items, iGrp - 1, 1);
            }
        }
        // Les couches concernées sont celles qui ont été préparées
        this.legendData = legendDataPrepared;
    },
    
    getTargetLegendData : function() {
        var legendDataPrepared = Ext.clone(this.legendData);
        var legendData = this.legendData;
        // Chaque groupe (à l'envers)
        for (iGrp = legendData.items.length; iGrp > 0; iGrp--) {
            grp = legendData.items[iGrp - 1];
            // Chaque couche (à l'envers)
            for(iLay=grp.items.length ; iLay>0 ; iLay--) {
                var layerDef = grp.items[iLay-1];
                if (!layerDef.visibility) {
                    // Pas de droit sur cette couche
                    Ext.Array.erase(legendDataPrepared.items[iGrp - 1].items, iLay - 1, 1);
                }
            }
        }
        // On retire les groupes vides
        for (iGrp = legendDataPrepared.items.length; iGrp > 0; iGrp--) {
            if (Ext.isEmpty(legendDataPrepared.items[iGrp - 1].items)) {
                Ext.Array.erase(legendDataPrepared.items, iGrp - 1, 1);
            }
        }
        // Les couches concernées sont celles qui ont été préparées
        return legendDataPrepared;
    },
    
    renderLegend : function() {
        // Légende
        if (this.legendTpl == null) {
            this.legendTpl = Ext.create('Sdis.Remocra.widget.map.LegendTemplate');
        }
        var legendNode = this.queryById('couches').getEl();
        this.legendTpl.overwrite(legendNode, {
            id: this.id,
            legendData: this.legendData
        }, false);
        
        // Mise à jour de la hauteur du conteneur
        Ext.defer(this.doLayout, 500, this);
    },
    
    getLayerByCode : function(layerCode) {
        var legendData = this.legendData, i, j;
        if (!legendData) {
            return null;
        }
        for (i=0 ; i<this.legendData.items.length ; i++) {
            var grp = this.legendData.items[i];
            for (j=0 ; j<grp.items.length ; j++) {
                var layerDef = grp.items[j];
                if (layerDef.id == layerCode) {
                    return layerDef;
                }
            }
        }
        return null;
    },
    setLayerVisibility : function(layerCode, visibility) {
        this.getLayerByCode(layerCode).visibility = visibility;
    },
    setLayerOpacity : function(layerCode, opacity) {
        this.getLayerByCode(layerCode).opacity = opacity;
    },
    
    getUrl : function() {
        var i, j;
        var titre = this.queryById('titre').getValue();
        var format = this.queryById('format').getValue();
        var orientation = this.queryById('orientation').getValue();
        var url = 'cartographie/carte/t/' + (encodeURIComponent(titre)||'Carte sans nom') + '/f/' + format + '/o/' + orientation;
        // Couches de la forme /{couche}/{opacité}
        for (i=0 ; i<this.legendData.items.length ; i++) {
            var grp = this.legendData.items[i];
            for (j=0 ; j<grp.items.length ; j++) {
                var layerDef = grp.items[j];
                if (layerDef.visibility && layerDef.opacity>0) {
                    url += '/'+layerDef.id+'/'+(layerDef.opacity||1);
                }
            }
        }
        return url;
    }
});